src/test/java/uk/gov/hmcts/reform/civil/utils/PartyUtilsTest.java

```diff
@@ -169,4 +172,52 @@ void shouldReturnEmpty_whenPartyTypeIsNotIndividualOrSoleTrader(Party.Type type)
             assertThat(PartyUtils.getDateOfBirth(party)).isEmpty();
         }
     }
+
+    @Nested
+    class PartyReferences {
+
+        @Test
+        void shouldReturnReferences_whenNoRefsAvailable() {
+            CaseData caseData = CaseDataBuilder.builder().build();
+
+            String partyReferences = PartyUtils.buildPartiesReferences(caseData);
+
+            assertEquals("", partyReferences);
+        }
+
+        @Test
+        void shouldReturnReferences_whenNot1v2DiffSolicitorAndBothRefAvailable() {
+            CaseData caseData = CaseDataBuilder.builder().atStateClaimDraft().build();
+
+            String partyReferences = PartyUtils.buildPartiesReferences(caseData);
+
+            assertEquals("Claimant reference: 12345\nDefendant reference: 6789", partyReferences);
+        }
+
+        @Test
+        void shouldReturnReferences_whenNot1v2DiffSolicitorAndOnlyClaimantRefAndSol2RefAvailable() {
```

Comment:
when not 1v2 different solicitor, case won't have solicitor 2.
Is this a 1v2DiffSolicitor case?

Reply:
As the name suggests, this is a scenario of 1v2 Different Solictors, but only claimant and defendant sol 2 refs are available. defendant ref 1 is not available. so in the outcome, we'll only expect the entries which are present. (notably all three refs are optional in this scenario, similarly the claimant and defendant 1 refs are optional in 1v1 case or 1v2 same sol case). I hope I understood the comment right.

Reply:
The name of test has been corrected